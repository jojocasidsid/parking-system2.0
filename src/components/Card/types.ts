export interface IProps {
	header: JSX.Element | JSX.Element[]
	children: JSX.Element | JSX.Element[]
}
