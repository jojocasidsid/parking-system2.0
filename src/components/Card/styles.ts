import styled from '@emotion/styled'

export const CardWrapper = styled('div')`
	display: flex;
	justify-content: flex-start;
	align-items: center;
	padding: 20px;
	flex-direction: column;
	border-radius: 30px;
	border: 2px solid ${({ theme }) => theme.palette.secondary.main};
`
export const StyledSlotHeader = styled('div')`
	display: flex;
	justify-content: space-around;
	align-items: center;
`
