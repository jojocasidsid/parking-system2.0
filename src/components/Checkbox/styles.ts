import styled from '@emotion/styled'

export const StyledRoot = styled('div')`
	padding: 18px;
`
