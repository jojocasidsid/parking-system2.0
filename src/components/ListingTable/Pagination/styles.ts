import styled from '@emotion/styled'
import { Typography } from '@mui/material'

export interface IPageInputProps {
	error?: string
}

export const PageInput = styled.input<IPageInputProps>`
	border-radius: 2px 0px 0px 2px;
	padding: 7px 2px 7px 0px;
	text-align: center;
	width: 100%;
	font-style: normal;
	font-size: 14px;
	line-height: 16px;
	font-weight: 500;
	font-stretch: normal;
	height: 28px;

	&::-webkit-outer-spin-button,
	&::-webkit-inner-spin-button {
		-webkit-appearance: none;
		margin: 0;
	}

	&:active,
	&:focus {
		border: 1px solid ${({ theme }) => theme.palette.primary.main};
		outline: none;
		box-shadow: none;
	}

	&:disabled {
		background-color: ${({ theme }) => theme.palette.text.secondary};
		pointer-events: none;
	}

	border: 1px solid
		${({ theme, error }) => {
			if (error) {
				return theme?.palette.error.main
			}
			return theme.palette.text.secondary
		}};
`

export const StyledTypography = styled(Typography)`
	color: #303643b3;
	font-weight: bold;
	font-size: 12px;
`
