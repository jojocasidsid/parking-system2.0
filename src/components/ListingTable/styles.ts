import styled from '@emotion/styled'
import { Typography } from '@mui/material'

export const StyledTypography = styled(Typography)`
	color: #30364380;
	font-weight: bold;
	font-size: 14px;
`
