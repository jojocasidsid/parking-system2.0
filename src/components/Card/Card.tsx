import React from 'react'
import { CardWrapper, StyledSlotHeader } from './styles'
import { IProps } from './types'

const Card = ({ header, children }: IProps) => (
	<CardWrapper>
		<StyledSlotHeader>{header}</StyledSlotHeader>
		<div>{children}</div>
	</CardWrapper>
)

export default Card
