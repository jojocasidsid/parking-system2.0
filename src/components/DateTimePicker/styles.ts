import styled from '@emotion/styled'
import { InputLabel, TextField } from '@mui/material'
import { Theme } from '@mui/material/styles'

interface ITheme {}
interface IInput extends ITheme {
	errors?: boolean
	theme?: Theme
}

export const StyledRoot = styled('div')`
	display: block;
`

export const StyledLabel = styled(InputLabel)`
	text-align: 'left';
	letter-spacing: 0.12px;
	padding-bottom: 8px;

	font-style: normal;
	font-size: 12px;
	line-height: 14px;
	font-weight: 600;
	font-stretch: normal;
	white-space: normal;
	//has error
	color: ${({ theme, errors }: IInput) => {
		if (errors) {
			return theme?.palette.error.main
		}
		return theme?.palette.text.primary
	}};
	& span {
		color: ${({ theme }) => theme.palette.error.main};
	}

	display: block;
`

export const StyledInput = styled(TextField)`
	input {
		padding: 8px 16px 8px 16px;
		font-style: normal;
		font-size: 14px;
		font-weight: 500;
		font-stretch: normal;
		color: ${({ theme }) => theme.palette.text.primary};
	}

	fieldset {
		border: 1px solid ${({ theme }) => theme.palette.text.secondary};
		border-radius: 2px 0px 0px 2px;
		border-color: ${({ theme, errors }: IInput) =>
			errors
				? theme?.palette.error.main
				: theme?.palette.text.secondary} !important;
	}

	input.Mui-disabled {
		background-color: ${({ theme }) => theme.palette.text.secondary};
		pointer-events: none;
	}

	input.MuiInputBase-inputAdornedStart {
		padding-left: 0px;
	}

	input.MuiInputBase-inputAdornedEnd {
		padding-right: 8px;
	}

	.MuiFormHelperText-root {
		color: ${({ theme }) => theme.palette.error.main};
		margin-left: 0;
	}
`
