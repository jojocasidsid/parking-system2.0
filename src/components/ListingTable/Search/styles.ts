import styled from '@emotion/styled'
import { InputLabel, TextField } from '@mui/material'
import { Theme } from '@mui/material/styles'

interface IInput {
	error?: boolean
	theme?: Theme
	small?: boolean
}

export const StyledRoot = styled('div')`
	display: block;
`

export const StyledLabel = styled(InputLabel)`
	text-align: 'left';
	letter-spacing: 0.12px;
	padding-bottom: 8px;

	font-style: normal;
	font-size: 12px;
	line-height: 14px;
	font-weight: 600;
	font-stretch: normal;
	white-space: normal;
	//has error
	color: ${({ theme, error }: IInput) => {
		if (error) {
			return theme?.palette.error.main
		}
		return theme?.palette.text.primary
	}};
	& span {
		color: ${({ theme }) => theme.palette.error.main};
	}

	display: block;
`

export const StyledInput = styled(TextField)`
	input {
		padding: ${({ small }: IInput) =>
			small ? '2px 8px 2px 8px' : '8px 12px 8px 16px'};
		font-style: normal;
		font-size: 14px;
		font-weight: 500;
		font-stretch: normal;
		color: ${({ theme }) => theme?.palette.text.primary};
	}

	fieldset {
		border: 1px solid ${({ theme }) => theme.palette.text.secondary};
		border-radius: 2px 0px 0px 2px;
	}

	input.Mui-disabled {
		background-color: ${({ theme }) => theme.palette.text.secondary};
		pointer-events: none;
	}

	input.MuiInputBase-inputAdornedStart {
		padding-left: 0px !important;
	}

	input.MuiInputBase-inputAdornedEnd {
		padding-right: 0px;
	}

	.MuiInputBase-adornedEnd div.MuiInputAdornment-root {
		background-color: ${({ theme }) => theme.palette.primary.contrastText};
		padding: 16px 16px 16px 16px;
		padding-right: 0px !important;
	}

	.MuiInputBase-adornedStart {
		padding-left: 0px !important;
	}

	.MuiInputBase-adornedStart .MuiInputAdornment-root {
		background-color: ${({ theme }) => theme.palette.primary.contrastText};
		padding: 16px 16px 16px 16px;
	}

	.MuiFormHelperText-root {
		margin-left: 0;
	}
`
