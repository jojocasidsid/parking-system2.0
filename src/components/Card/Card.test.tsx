import React from 'react'
import renderer from 'react-test-renderer'

// packages
import { ThemeProvider } from '@mui/material/styles'

// config
import theme from 'config/theme'

// component
import Card from './Card'

describe('Card Component', () => {
	it('should render card component without crashing', () => {
		const tree = renderer.create(
			<ThemeProvider theme={theme}>
				<Card header={<h1>Hello World</h1>}>
					<h1>Hello World</h1>
				</Card>
			</ThemeProvider>
		)

		expect(tree).toMatchSnapshot()
	})
})
