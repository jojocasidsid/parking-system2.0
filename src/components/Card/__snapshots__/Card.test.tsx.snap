// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component should render card component without crashing 1`] = `
<div
  className="css-joqnyj"
>
  <div
    className="css-e98rco"
  >
    <h1>
      Hello World
    </h1>
  </div>
  <div>
    <h1>
      Hello World
    </h1>
  </div>
</div>
`;
